### PR TITLE
fix: add error handling to BaseAudioContext::new_inherited

### LINKED ISSUE #32986: Panic in webaudio/the-audio-api/the-audiocontext-interface/audiocontextoptions.html
To reproduce: `./mach test-wpt tests/wpt/tests/webaudio/the-audio-api/the-audiocontext-interface/audiocontextoptions.html`

Backtrace:
```
pid:47743 Could not start audio render thread: Backend("AudioInfo failed: BoolError { message: \"Failed to build AudioInfo\", filename: \"/Users/jdm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gstreamer-audio-0.22.6/src/audio_info.rs\", function: \"gstreamer_audio::audio_info::AudioInfoBuilder::build\", line: 77 }") (thread AudioRenderThread, at /Users/jdm/.cargo/git/checkouts/media-c23a3cd5aa97076a/45756be/audio/render_thread.rs:165)
 0:05.68 pid:47743   13: core::result::Result<T,E>::expect
 0:05.68 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/result.rs:1059:23
 0:05.69 pid:47743   14: servo_media_audio::render_thread::AudioRenderThread::start
 0:05.69 pid:47743              at /Users/jdm/.cargo/git/checkouts/media-c23a3cd5aa97076a/45756be/audio/render_thread.rs:164:26
 0:05.70 pid:47743   15: servo_media_audio::context::AudioContext::new::{{closure}}
 0:05.70 pid:47743              at /Users/jdm/.cargo/git/checkouts/media-c23a3cd5aa97076a/45756be/audio/context.rs:149:17
 0:05.71 pid:47743   16: std::sys_common::backtrace::__rust_begin_short_backtrace
 0:05.71 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:155:18
 0:05.71 pid:47743   17: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
 0:05.71 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/thread/mod.rs:542:17
 0:05.72 pid:47743   18: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
 0:05.72 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/panic/unwind_safe.rs:272:9
 0:05.73 pid:47743   19: std::panicking::try::do_call
 0:05.74 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
 0:05.74 pid:47743   20: ___rust_try
 0:05.74 pid:47743   21: std::panicking::try
 0:05.74 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
 0:05.74 pid:47743   22: std::panic::catch_unwind
 0:05.74 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
 0:05.74 pid:47743       std::thread::Builder::spawn_unchecked_::{{closure}}
 0:05.74 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/thread/mod.rs:541:30
 0:05.75 pid:47743   23: core::ops::function::FnOnce::call_once{{vtable.shim}}
 0:05.75 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:250:5
 0:05.75 pid:47743   24: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
 0:05.75 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/alloc/src/boxed.rs:2063:9
 0:05.75 pid:47743       <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
 0:05.75 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/alloc/src/boxed.rs:2063:9
 0:05.75 pid:47743       std::sys::pal::unix::thread::Thread::new::thread_start
 0:05.75 pid:47743              at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys/pal/unix/thread.rs:108:17
 0:05.75 pid:47743   25: __pthread_joiner_wake
```

### PR DESCRIPTION
**Pull request by Taym95**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #32986

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

File: components/script/dom/audiocontext.rs [hunk 1]
```diff
@@ -48,12 +48,15 @@ pub struct AudioContext {
 impl AudioContext {
     #[allow(crown::unrooted_must_root)]
     // https://webaudio.github.io/web-audio-api/#AudioContext-constructors
-    fn new_inherited(options: &AudioContextOptions, pipeline_id: PipelineId) -> AudioContext {
+    fn new_inherited(
+        options: &AudioContextOptions,
+        pipeline_id: PipelineId,
+    ) -> Fallible<AudioContext> {
```
Comment (Taym95) [inline components/script/dom/audiocontext.rs:54]: @gterzian @jdm this is ready for review and depends on https://github.com/servo/media/pull/424

**Review by jdm — APPROVED** [review]
Nice work!

**Comment by Taym95** [comment]
> Test results for linux-wpt-layout-2020 from try job ([#10369821790](https://github.com/servo/servo/actions/runs/10369821790)):
> 
> Flaky unexpected result (20)
> Stable unexpected results that are known to be intermittent (11)
> Stable unexpected results (1)
> * TIMEOUT [expected OK] `/fetch/metadata/audio-worklet.https.html`

Interesting, how can test timeout and I am returning error now?!

**Comment by jdm** [comment]
Have you tested the original test from #32986 locally? I'd like to know why it's still crashing.

**Comment by Taym95** [comment]
> Have you tested the original test from #32986 locally? I'd like to know why it's still crashing.

Yes it is not crashing but timeout:

```
0:02.80 TEST_START: /webaudio/the-audio-api/the-audiocontext-interface/audiocontextoptions.html
 0:02.80 WARNING Got command init_succeeded in state running
 0:27.80 INFO Got timeout in harness
 0:27.80 TEST_END: TIMEOUT, expected CRASH - TestRunner hit external timeout (this may indicate a hang)
 0:27.80 INFO Pausing until the browser exits
 ```

**Comment by Taym95** [comment]
Great news, it's passing now! The issue was in another PR—`receiver` was blocking:


image

File: Cargo.toml [hunk 2]
```diff
@@ -101,9 +101,9 @@ selectors = { git = "https://github.com/servo/stylo", branch = "2024-07-16" }
 serde = "1.0.205"
 serde_bytes = "0.11"
 serde_json = "1.0"
-servo-media = { git = "https://github.com/servo/media" }
-servo-media-dummy = { git = "https://github.com/servo/media" }
-servo-media-gstreamer = { git = "https://github.com/servo/media" }
+servo-media = { git = "https://github.com/Taym95/media", rev = "5c16fdf2d97c1540586e8dd730c02db7753b72b0" }
```
Comment (Taym95) [inline Cargo.toml:104]: merge this first

**Comment by gterzian** [comment]
https://github.com/servo/servo/actions/runs/10402430018/job/28806906443#step:11:35

**Review by jdm — APPROVED** [review]

**Pull request merged**